### PR TITLE
4079 - Add tests for createWagmiConfig

### PR DESCRIPTION
**Pull request by w1that**
[ISSUE](https://github.com/0xsequence/issue-tracker/issues/4079)

This PR introduces a comprehensive test suite for the createWagmiConfig function, covering both successful and failure scenarios. The tests ensure correct behavior across different wallet configurations and edge cases.

Changes:
	•	Added test cases for creating configs with:
	•	Empty collections (defaulting to Polygon)
	•	Universal wallet setup
	•	Embedded wallet setup
	•	EIP6963 wallet inclusion/exclusion
	•	SSR compatibility
	•	Handling missing waasConfigKey (falling back to universal wallet)
	•	Added failure cases to ensure proper error handling:
	•	Using embedded wallets without a valid waasConfigKey
	•	Creating config without walletConnectProjectId
	•	Creating config with an empty projectAccessKey

### CODE REVIEW

File: packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+					walletType: 'universal',
+				},
+			};
+
+			const config = createWagmiConfig(marketplaceConfig, baseSdkConfig);
+			expect(config.connectors).toBeDefined();
+			expect(config.chains).toHaveLength(1);
+		});
+
+		it('should still create config when projectAccessKey is empty', () => {
```
Comment (w1that) [inline packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts:186]: Current behaviour for `sdkConfig` without `projectAccessKey` is this. We need to check if it's an empty string and throw an error accordingly IMO, wdyt?

Comment (AlexanderKolberg) [reply packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts:186]: I think you are right we should just throw an error if the access key is undefined

File: packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			expect(config.chains).toHaveLength(1);
+		});
+
+		it('should create SSR compatible config when ssr flag is true', () => {
+			const config = createWagmiConfig(
+				baseMarketplaceConfig,
+				baseSdkConfig,
+				true,
+			);
+			expect(config.storage).toBeDefined();
```
Comment (AlexanderKolberg) [inline packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts:129]: Can we bee a bit more specific? this needs to be a cookie storage for SSR to work, can we identify that?

Comment (w1that) [reply packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts]: thanks for review! [I refactored it here](https://github.com/0xsequence/marketplace-sdk/pull/156/commits/05fea0043f4e089dcb646a041d1361885f4ca44a)

File: packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts (hunk 1 above)
Comment (w1that) [reply packages/sdk/src/react/_internal/wagmi/__tests__/create-config.test.ts]: I chose to handle it on Provider level, last 3 commits are about tests for Provider 👇🏼

**Review by AlexanderKolberg — APPROVED** [review]
LGTM!

Good work
